### PR TITLE
[SR] Tabs Block (Wave 3)

### PR DESCRIPTION
**Pull request by ivanvatadobe**
This PR adds the tabs block needed for ACE1209 (Site redesign wave 3) test. 

Note: this is the first push to `site-redesign-foundation`, and **_I would like to push this PR through immediately._** There are still items to implement, but I want to at least have a draft version of the wave 3 tabs in our shared demo pages. 

**Features:** 
- **Add variant: radio** - radio variant is re-enabled
- **Add variant: quiet** - basic text with blue underline as clickable tabs
- **Add dark mode for base variant (pill)**
- **Add dark mode for radio variant**
- **Add dark mode for quiet variant** 

**TODO/Notes**
1. confirm spacing for top level tabs when nested (section spacing is not applied below the parent tab).
2. confirm left/right mode works as intended from ace1205.
3. implement autoscroll to center selected radio tab in lower breakpoints (this is already available for base pill). 

**Resolves:** 
[MWPW-198998](https://jira.corp.adobe.com/browse/MWPW-198998) 

**Test URLs:**
- Before: https://main--da-cc--adobecom.aem.page/cc-shared/fragments/tests/2026/q3/ace1209/playground/viloria/tabs-layers-poc-radio-dark?milolibs=site-redesign-foundation&martech=off
- After: https://main--da-cc--adobecom.aem.page/cc-shared/fragments/tests/2026/q3/ace1209/playground/viloria/tabs-layers-poc-pill-dark?milolibs=sr3-tabs-block-radio&martech=off

**DEMO PAGES WITH TABS BY TYPE AND MODE** 

**base pill - default**
https://main--da-cc--adobecom.aem.page/cc-shared/fragments/tests/2026/q3/ace1209/playground/viloria/tabs-layers-poc-pill-light?milolibs=sr3-tabs-block-radio

**base pill - dark**
https://main--da-cc--adobecom.aem.page/cc-shared/fragments/tests/2026/q3/ace1209/playground/viloria/tabs-layers-poc-pill-dark?milolibs=sr3-tabs-block-radio

**radio - default**
https://main--da-cc--adobecom.aem.page/cc-shared/fragments/tests/2026/q3/ace1209/playground/viloria/tabs-layers-poc-radio-light?milolibs=sr3-tabs-block-radio

**radio - dark**
https://main--da-cc--adobecom.aem.page/cc-shared/fragments/tests/2026/q3/ace1209/playground/viloria/tabs-layers-poc-radio-dark?milolibs=sr3-tabs-block-radio

### CODE REVIEW

**Comment by rgclayton** [comment]
Clicking any of the parent tabs (photo, design, video) causes the page to jump up.

File: libs/mep/ace1209/tabs/tabs.css [hunk 1]
```diff
@@ -13,6 +15,157 @@
...
+          background: transparent;
+          color: var(--s2a-color-content-default);
+          border-bottom-color: var(--s2a-color-blue-800);
+        }
+      }
+    }
+  }
+
+  &.dark:not(.radio):not(.quiet) > .tabs-wrapper {
+    [role='tablist'] {
```
Comment (rgclayton) [inline libs/mep/ace1209/tabs/tabs.css:241]: This and on line 165 — the condition `:not(.radio):not(.quiet)` is duplicated across two separate rule blocks; could be consolidated (e.g., a shared class or comment noting they must stay in sync) to reduce future drift risk.

**Comment by rgclayton** [comment]
Looks like some alignment issues on mobile compared [to the design](https://www.figma.com/design/brgxjoNmioT7z6zFk1QHbR/Plans-%E2%80%94-A.com?node-id=2918-20714&m=dev)
image

image

File: libs/mep/ace1209/tabs/tabs.js [hunk 2]
```diff
@@ -271,39 +266,48 @@ const init = async (block) => {
...
       const tabBtn = createTag('button', tabBtnAttributes, item.textContent);
       const btnWrapper = createTag('div', { class: 'btn-wrapper' });
       btnWrapper.append(tabBtn);
       tabListContainer.append(btnWrapper);
 
       const tabContentAttributes = {
-        id: `tab-panel-${tabId}-${tabName}`,
-        role: 'tabpanel',
+        id: controlId,
+        ...(isRadio ? {} : { role: 'tabpanel' }),
```
Comment (zagi25) [inline libs/mep/ace1209/tabs/tabs.js:309]: **Blocking (radio visual regression):** radio panels get no role here, so they render as `<div class="tabpanel">`. But the layout rules in `tabs.css:295` and `:357-359` still key off `[role='tabpanel']` (section positioning, content centering, background z-index), and content `.section`s are appended into these panels — so the radio variant won't receive them → visual regression. Fix: either keep `role='tabpanel'` on radio panels, or update those two CSS selectors to `.tabpanel` (matching the change you already made at line 377).

File: libs/mep/ace1209/tabs/tabs.js [hunk 3]
```diff
@@ -271,39 +266,48 @@ const init = async (block) => {
...
+        id: btnId,
         tabindex: (i === 0) ? '0' : '-1',
-        'aria-selected': (i === 0) ? 'true' : 'false',
+        [isRadio ? 'aria-checked' : 'aria-selected']: (i === 0) ? 'true' : 'false',
         'data-block-id': `tabs-${tabId}`,
         'daa-state': 'true',
-        'daa-ll': `tab-${tabId}-${tabName}`,
-        'aria-controls': controlId,
+        'daa-ll': btnId,
+        ...(isRadio ? { 'data-control-id': controlId } : { 'aria-controls': controlId }),
```
Comment (zagi25) [inline libs/mep/ace1209/tabs/tabs.js:300]: **Radio a11y:** radio buttons use `data-control-id` instead of `aria-controls`, and the panels are role-less (line 308), so there's no programmatic tab→panel association — a screen-reader user isn't told that selecting a radio revealed content elsewhere, and the `aria-labelledby` on the role-less panel (line 310) is effectively ignored by AT. Somewhat inherent to modeling a tab/disclosure widget as a radiogroup — worth tracking with the radio TODOs.

File: libs/mep/ace1209/tabs/tabs.js [hunk 4]
```diff
@@ -191,8 +183,8 @@ function configTabs(config, rootElem) {
 }
 
 function initTabs(elm, config, rootElem) {
-  const tabs = elm.querySelectorAll('[role="tab"]');
-  const tabLists = elm.querySelectorAll('[role="tablist"]');
+  const tabs = elm.querySelectorAll(':scope > .tabs-wrapper [role="tab"], :scope > .tabs-wrapper [role="radio"]');
+  const tabLists = elm.querySelectorAll(':scope > .tabs-wrapper [role="tablist"], :scope > .tabs-wrapper [role="radiogroup"]');
   let tabFocus = 0;
   tabLists.forEach((tabList) => {
     tabList.addEventListener('keydown', (e) => {
```
Comment (zagi25) [inline libs/mep/ace1209/tabs/tabs.js:190]: **Radio a11y (keyboard):** for `role=radiogroup`, APG expects arrow keys to move focus *and* check the radio (selection follows focus), and to honor Up/Down as well as Left/Right. Here arrows only move roving focus; `aria-checked` changes on click/Space/Enter, and only Left/Right are handled. Functional, but not the standard radiogroup interaction.

File: libs/mep/ace1209/tabs/tabs.js [hunk 5]
```diff
@@ -271,39 +266,48 @@ const init = async (block) => {
...
-  if (tabListLabel) tabList.setAttribute('aria-label', tabListLabel);
 
-  const tabListItems = rows[0].querySelectorAll(':scope li');
+  const tabListLabelEl = tabListContainer.querySelector(':scope > p');
+  if (tabListLabelEl && isRadio) {
+    tabListLabelEl.classList.add('tab-list-label', 'label');
+    tabListLabelEl.id = `tab-list-label-${tabId}`;
+    tabList.setAttribute('aria-labelledby', tabListLabelEl.id);
+  } else {
+    tabListLabelEl?.remove();
```
Comment (zagi25) [inline libs/mep/ace1209/tabs/tabs.js:280]: Nit (behavior change): for non-radio variants a stray `<p>` in the tab-list container is now removed; previously it would have survived. Looks intentional — just flagging.

File: libs/mep/ace1209/tabs/tabs.css [hunk 6]
```diff
@@ -1,5 +1,7 @@
+/* stylelint-disable no-descending-specificity */
```
Comment (zagi25) [inline libs/mep/ace1209/tabs/tabs.css:1]: Nit: blanket-disabling `no-descending-specificity` for the whole file is a smell — the ordering could be resolved instead of globally disabled.

**Review by zagi25 — COMMENTED** [review]
_Reviewed by Claude (Claude Code, Opus 4.8), run by @zagi25._

Solid, net-positive change for a flag-gated MEP experiment (dark/pill base + `radio` + `quiet`; migration off hardcoded hexes and `-knockout` tokens to valid semantic `--s2a-*` tokens — verified, no typos). The base/quiet variants meet the WAI-ARIA tabs pattern.

**One issue to fix before the radio variant carries real content** (inline on `tabs.js:308`): radio panels lose `role='tabpanel'` but the CSS at `tabs.css:295` / `:357-359` still targets `[role='tabpanel']` → visual regression.

**Radio-variant a11y deviations** (inline): selection doesn't follow focus + Up/Down unsupported; no programmatic panel association; and the radiogroup's accessible name depends on an authored `<p>` label or `config.pretext` — none if neither is present. Somewhat inherent to the radiogroup modeling; suggest tracking against the existing radio TODOs rather than blocking.

**Minor:** the legacy `.tabs-background-transparent` hook (`tabs.css:316-321`) only targets `[role='tablist']`, so it won't make the radio `role=radiogroup` transparent (edge case).

**Security:** no new risk — no unsafe sinks; the one dynamic `#`-selector uses a sanitized, code-generated id.
**Tests:** none added — consistent with the `ace*` experiment convention (a gap, not a regression given it's flag-gated).

File: libs/mep/ace1209/tabs/tabs.js (hunk 5 above)
Comment (ivanvatadobe) [reply libs/mep/ace1209/tabs/tabs.js:280]: this is intentional. the `<p>` determines the label for radio buttons. if radio is not enabled, then the p is deleted.

File: libs/mep/ace1209/tabs/tabs.js (hunk 3 above)
Comment (ivanvatadobe) [reply libs/mep/ace1209/tabs/tabs.js:300]: adding to todo. i will reach out for a general guidance on which solution to implement regarding `tabpanel` vs `region`.

File: libs/mep/ace1209/tabs/tabs.js (hunk 2 above)
Comment (ivanvatadobe) [reply libs/mep/ace1209/tabs/tabs.js:309]: updated using `.tabpanel` css selector option.

File: libs/mep/ace1209/tabs/tabs.js (hunk 4 above)
Comment (ivanvatadobe) [reply libs/mep/ace1209/tabs/tabs.js:190]: adding to todo. closely related with decision for item about (re `tabpanel` vs `region`). 

current behavior: 
left/right arrows - moves focus (does not activate) to tab in current tabgroup. 
down/up - does not change between tabgroups
tab - switches to next group/exists the tabgroups altogether when reaching the last child tab items
cmd+tab - switches back to the previous tabgroup
enter - activates focused tab in tabground

File: libs/mep/ace1209/tabs/tabs.css (hunk 1 above)
Comment (ivanvatadobe) [reply libs/mep/ace1209/tabs/tabs.css:241]: added `.pill` class to base variant to simplify selectors and removed the dupes.

**Comment by ivanvatadobe** [comment]
> Looks like some alignment issues on mobile compared [to the design](https://www.figma.com/design/brgxjoNmioT7z6zFk1QHbR/Plans-%E2%80%94-A.com?node-id=2918-20714&m=dev) image
> 
> image

added styling to fix this issue.

**Comment by ivanvatadobe** [comment]
@zagi25 

i met with @narcis-radu to discuss the three issues not yet addressed: 

1. APG expects arrow keys to move focus and check the radio (selection follows focus), and to honor Up/Down as well as Left/Right.
2. aria-labelledby on the role-less panel (line 310) is effectively ignored by AT.
3. page jump issue (also an issue in production on /acrobat.html). 

i am going to reach out to Taryn to discuss the accessibility issues- if she requests changes, then it will be listed as a follow-up, not required for the test unless absolutely necessary. 

I'm also going to investigate the page jump issue, but i don't believe that's a deal breaker for getting this pr approved for further testing on our test pages. 

please review the other changes when ready so i can get this to `site-redesign-foundation`.

**Comment by rgclayton** [comment]
> Clicking any of the parent tabs (photo, design, video) causes the page to jump up.

I had this happen today while testing. It happened clicking on a nested tab. 

I also wanted to verify if the merch cards in the nested tabs are all the same? When i click any of those tabs the content doesn't change, and the animation doesn't run. Is that expected?

**Comment by ivanvatadobe** [comment]
> > Clicking any of the parent tabs (photo, design, video) causes the page to jump up.
> 
> I had this happen today while testing. It happened clicking on a nested tab.
> 
> I also wanted to verify if the merch cards in the nested tabs are all the same? When i click any of those tabs the content doesn't change, and the animation doesn't run. Is that expected?

no... this isn't expected. the only issue (which is also replicable in production right now in c1) is the page jumping a bit when you click on the tabs when the page is not fully loaded yet (the bottom of the page takes a _very_ long time to load. 

the animations work as intended/accepted in the previous wave for me.

**Review by rgclayton — APPROVED** [review]

**Review by zagi25 — APPROVED** [review]

**Comment by DKos95** [comment]
I would suggest asking the accessibility team if the radio buttons are big enough to pass a11y, as far as I remember 24x24px is the minimum dimensions a clickable object can be and currently the radio buttons are 20x20.

**Review by narcis-radu — APPROVED** [review]

**Review by overmyheadandbody — APPROVED** [review]
Visual nit: the "g" in "Design" gets cut off at the bottom on desktop

Image

**Comment by ivanvatadobe** [comment]
> Visual nit: the "g" in "Design" gets cut off at the bottom on desktop
> 
> Image

i added css fixes for this issue.

**Review by SilviuLCF — APPROVED** [review]
Verified, testing details https://jira.corp.adobe.com/browse/MWPW-198998
